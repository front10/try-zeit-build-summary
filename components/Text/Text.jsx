import React from 'react';
import PropTypes from 'prop-types';
import useStyles from '@front10/helpers/dist/UseStyles';

const Text = props => {
  const {
    children,
    id,
    type,
    htmlFor,
    ariaLabel,
    ariaHidden,
    role,
    dataReactCmpOwner,
    onClick
  } = props;
  const Tag = type;
  const dataAttrs = dataReactCmpOwner ? { 'data-react-cmp-owner': dataReactCmpOwner } : {};
  const [className, style] = useStyles(props);

  return (
    <Tag
      className={`fr-${type} ${className}`}
      id={id}
      style={style}
      htmlFor={htmlFor}
      aria-label={ariaLabel}
      aria-hidden={ariaHidden}
      role={role}
      onClick={onClick}
      {...dataAttrs}
    >
      {children}
    </Tag>
  );
};

Text.propTypes = {
  /**
   * id of components
   */
  id: PropTypes.string,
  /**
   * Icon to show in component
   */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /**
   * Type of the element
   */
  type: PropTypes.oneOf(['span', 'label', 'p']),
  /**
   * Defne htmlFor
   */
  htmlFor: PropTypes.string,
  /**
   * aria label
   */
  ariaLabel: PropTypes.string,
  /**
   * hide content to screen readers
   */
  ariaHidden: PropTypes.bool,
  /**
   * Role atribute
   */
  role: PropTypes.string,
  /**
   * Allows the creation of a data-react-cmp-owner data-attr
   */
  dataReactCmpOwner: PropTypes.string,
  /**
   * onClick function
   */
  onClick: PropTypes.func
};
Text.defaultProps = {
  id: null,
  children: null,
  type: 'span',
  htmlFor: null,
  ariaLabel: null,
  ariaHidden: null,
  role: null,
  dataReactCmpOwner: null,
  onClick: null
};

Text.componentName = 'Text';

export default Text;
