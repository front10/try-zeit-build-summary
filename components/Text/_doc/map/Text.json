{"description":"","displayName":"Text","methods":[],"props":{"className":{"type":{"name":"string"},"required":false,"description":"Class name of span","defaultValue":{"value":"''","computed":false}},"style":{"type":{"name":"objectOf","value":{"name":"any"}},"required":false,"description":"Css style applied to the component","defaultValue":{"value":"null","computed":false}},"id":{"type":{"name":"string"},"required":false,"description":"id of components","defaultValue":{"value":"null","computed":false}},"children":{"type":{"name":"union","value":[{"name":"arrayOf","value":{"name":"node"}},{"name":"node"}]},"required":false,"description":"Icon to show in component","defaultValue":{"value":"null","computed":false}},"type":{"type":{"name":"enum","value":[{"value":"'span'","computed":false},{"value":"'label'","computed":false},{"value":"'p'","computed":false}]},"required":false,"description":"Type of the element","defaultValue":{"value":"'span'","computed":false}},"htmlFor":{"type":{"name":"string"},"required":false,"description":"Defne htmlFor","defaultValue":{"value":"null","computed":false}},"ariaLabel":{"type":{"name":"string"},"required":false,"description":"aria label","defaultValue":{"value":"null","computed":false}},"ariaHidden":{"type":{"name":"bool"},"required":false,"description":"hide content to screen readers","defaultValue":{"value":"null","computed":false}},"role":{"type":{"name":"string"},"required":false,"description":"Role atribute","defaultValue":{"value":"null","computed":false}},"dataReactCmpOwner":{"type":{"name":"string"},"required":false,"description":"Allows the creation of a data-react-cmp-owner data-attr","defaultValue":{"value":"null","computed":false}}},"filename":"/Users/darielnoel/repositories/front10/airmodules/src/components/elements/Text/Text.jsx"}