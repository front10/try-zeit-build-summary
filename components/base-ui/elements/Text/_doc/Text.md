# Text
  [![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)
  

  ## Properties
  | </br>Name | </br>Type | </br>Summary | </br>Default | 
| ---- | ---- | ---- | ---- |
| ariaHidden | `bool` | hide content to screen readers | `null` |
| ariaLabel | `string` | aria label | `null` |
| children | `union` | Icon to show in component | `null` |
| dataReactCmpOwner | `string` | Allows the creation of a data-react-cmp-owner data-attr | `null` |
| htmlFor | `string` | Defne htmlFor | `null` |
| id | `string` | id of components | `null` |
| onClick | `func` | onClick function | `null` |
| role | `string` | Role atribute | `null` |
| type | `enum` | Type of the element | `'span'` |