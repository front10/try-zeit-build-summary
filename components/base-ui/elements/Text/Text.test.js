import React from 'react';
import { render, waitForElement } from 'test-utils';
import Text from './Text';


test('render and match with snapshot', () => {
  const { container } = render(<Text />);
  expect(container).toMatchSnapshot();
});

