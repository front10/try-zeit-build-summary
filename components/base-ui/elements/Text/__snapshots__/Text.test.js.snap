// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render all 1`] = `
<span
  aria-hidden={null}
  aria-label={null}
  className="fr-span "
  htmlFor={null}
  id={null}
  role={null}
  style={
    Object {
      "filter": undefined,
    }
  }
/>
`;
